Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -252,9 +252,17 @@ CancellationToken cancellationToken
                         var properties = model.CreateBasicProperties();
                         rawMessage.Properties.CopyTo(properties);
                         var confirmation = confirmationListener.CreatePendingConfirmation(model);
-                        model.BasicPublish(exchange.Name, routingKey, mandatory, properties, rawMessage.Body);
+                        try
+                        {
+                            model.BasicPublish(exchange.Name, routingKey, mandatory, properties, rawMessage.Body);
+                        }
+                        catch (Exception)
+                        {
+                            confirmation.Cancel();
```

Comment:
I have returned this logic because otherwise in case of merely long downtime and huge RPS OOM could be thrown.